### DISCUSSION TITLE
NextJS's JSX Transform's add-missing-react-import codemod contradicts  React's removing-unused-react-imports codemod

### DISCUSSION THREAD

**Discussion by TriStarGod** [vercel/next.js#24877] [Help]
On NextJS's doc [add-missing-react-import](https://nextjs.org/docs/advanced-features/codemods#add-missing-react-import) , it mentions to run the codemod to add `import React from "react";` for JSX Transform capabilities. However, on ReactJS's blog [removing-unused-react-imports](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports) , they mention removing the "extra"` import React` if its found in the file since its unnecessary for JSX Transforms. Which one is correct?

**Comment by rporrasluc** [comment]
I might be wrong, but I came here because of having the same doubts. I can imagine that previously Next.js was importing react in all files using JSX under the hood in order to use `React.createElement`. As for the changes in react and the new transform `React.createElement` is not anymore used and `react/jsx-runtime` is so I assume Next.js is importing that one under the hood now.

The `codemod` is needed in case you were using other React functions (like React.Component in the example) which were working without importing React due to Next.js doing that by default.
